connector/mqtt/vernemq/v2k-bridge/README.md

```diff
@@ -16,84 +16,86 @@ secured with mutual TLS. In future releases, the communication with Kafka will a
+Before running the V2K Bridge service within your environment, make sure you configure the variables
+to match your needs.
+
+You can select the configuration file via the `V2K_APP_USER_CONFIG_FILE` variable. Its default value
+is `production.conf`.
+
+For more information about the usage of the configuration files and environment variables, check the
+__ConfigManager__ module in our[Microservice SDK](https://github.com/dojot/dojot-microservice-sdk-js).
+
+## **App**
+
+Configurations used in the service.
+
+| Key | Purpose | Default Value | Valid Values
+| --- | ------- | ------------- | ------------
+| backpressure.handlers | Number of parallel handlers for the backpressure queue processing mechanism | 4 | integer
+| backpressure.queue.length.max | Maximum backpressure queue length in bytes | 1048576 | integer
+| messenger.produce.topic.suffix | Kafka production topic suffix | device-data | string
+| logger.transports.console.level | Console logger level | info | string
+| logger.verbose | Whether to enable logger verbosity or not | false | boolean
+| subscription.qos | QoS to be used when subscribing to the MQTT broker | 1 | _0_, _1_, _2_
+| subscription.topic | Subscription topic | $share/group/+/attrs | string
+
+## **MQTT**
+
+Configurations passed directly to the [NPM MQTT library](https://www.npmjs.com/package/mqtt). We do
+not set the default values for all of them, but you can change the value of the ones we did not
+mentioned here via environment variables.
+
+Check the documentation for more information on these variables.
+
+__NOTE THAT__ the module adapts the dotted version of the variables to the camelCase one that the
+library accepts.
+
+| Key | Default Value | Valid Values
+| --- | ------------- | ------------
+| mqtt.ca                  | /opt/v2k_bridge/app/cert/ca.crt                      | path
+| mqtt.cert                | /opt/v2k_bridge/app/cert/${HOSTNAME:-v2k-bridge}.crt | path
+| mqtt.clean               | false                                                | boolean
+| mqtt.client.id           | ${HOSTNAME:-v2k-bridge}                              | string
+| mqtt.host                | vernemq-k8s                                          | string
+| mqtt.keep.alive           | 60                                                   | integer
+| mqtt.key                 | /opt/v2k_bridge/app/cert/${HOSTNAME:-v2k-bridge}.key | path
+| mqtt.port                | 8883                                                 | integer
+| mqtt.protocol            | mqtts                                                | _mqtt_ and _mqtts_
+| mqtt.reject.unauthorized | true                                                 | boolean
+| mqtt.username            | ${HOSTNAME:-v2k-bridge}                              | string
+
+## **SDK Producer**
+
+These parameters are passed directly to the SDK producer. Check the
+[official repository](https://github.com/dojot/dojot-microservice-sdk-js) for more info on the
+values.
```

Comment:
Just commenting on a point of view:
I really like this idea of passing the settings directly to the components. At first I even thought about adopting this same strategy, but then I rethought and decided to filter all the parameters that were passed from the `config` object to the components.
Since I don't have a handle on the security holes that this approach can open, I preferred to keep track of everything that can be passed on to the components. This way I can avoid injecting settings that are not provided for in the documentation.